test/testerrorlogger.cpp

```diff
@@ -301,15 +301,15 @@ class TestErrorLogger : public TestFixture {
         errout.str("");
         suppressions.clear();
         suppressions.emplace_back("abc", "a.c", 10U);
-        suppressions.emplace_back("unmatchedSuppression", "*", 0U);
+        suppressions.emplace_back("unmatchedSuppression", "*", /*Suppressions::Suppression::NO_LINE*/-1); //TODO find out why clang can't link with Suppressions::Suppression::NO_LINE
```

Comment:
The problem you had was a linker error because there is no definition for `Suppressions::Suppression::NO_LINE`.

I would guess that if you replace the variable with such constant it will work:

    enum {NO_LINE=-1};

Another solution is to add a variable definition.. but that is unfortunate.

Reply:
I want that you _replace_ the variable declaration in suppression.h

This line in suppressions.h:

    static const int NO_LINE = -1;

Write this instead:

    enum { NO_LINE = -1 };

The code says that there is a constant variable with the name NO_LINE but it does not define the variable. The compiler expects that the variable is defined elsewhere. The error should also go away if you add this in a source file:

    const int Suppressions::Suppression::NO_LINE;

But I think it is unfortunate to use a constant variable because then when the compiler compiles testerrorlogger it will not _know_ if the value is -1 so it has to add code that loads the variable. The value of the enum constant would be known so the compiler can just insert `-1` if that is used.

Reply:
As I read it on this page:
https://en.cppreference.com/w/cpp/language/static

the code is not valid. you need to define `Suppressions::Suppression::NO_LINE` also.